### PR TITLE
[WIP] add Cudnn mode as optional param to batch norm and softmax

### PR DESCRIPTION
**Pull request by dcslin**

### CODE REVIEW

**Comment by dcslin** [comment]
split to https://github.com/apache/incubator-singa/pull/540 and https://github.com/apache/incubator-singa/pull/541
